src/arrayMethodSort.js

```diff
@@ -4,8 +4,29 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = function (a, b) {
+      return (
+        String(a).toLowerCase().localeCompare(String(b).toLowerCase()) ||
+        String(a).localeCompare(String(b)) ||
+        String(a)
+          .toLocaleUpperCase()
+          .localeCompare(String(b).toLocaleUpperCase())
+      );
+    },
+  ) {
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compareFunction(this[j], this[j + 1]) > 0) {
+          const temp = this[j];
+
+          this[j] = this[j + 1];
+          this[j + 1] = temp;
+        }
+      }
+    }
```

Comment:
You've implemented a bubble sort algorithm which is correct and meets the task's requirements. However, keep in mind that bubble sort has O(n^2) time complexity, which is not efficient for large arrays. For better performance on larger data sets, consider researching more efficient sorting algorithms like quicksort or mergesort.